.github/workflows/deploy-to-test-and-live.yml

```diff
@@ -0,0 +1,78 @@
+# Github action to deploy to Pantheon
+name: Deploy to Pantheon Test and Live
+
+on:
+  release:
+    types: [published]  # Trigger when a release is published
+
+jobs:
+  deploy-to-test:
+    environment: Test
+    permissions:
+      contents: read
+      deployments: write
+      pull-requests: read
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.release.tag_name }}  # Checkout the release tag (e.g., v1.0.0)
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "7.4"
+
+      - name: Install Terminus
+        uses: pantheon-systems/terminus-github-actions@v1
+        with:
+          pantheon-machine-token: ${{ secrets.PANTHEON_MACHINE_TOKEN }}
+          disable-cache: true
+
+      - name: Prepare deployment message
+        run: |
+          RELEASE_BODY=$(echo "${{ github.event.release.body }}")
+          echo "DEPLOY_TEST_MESSAGE<<EOF" >> $GITHUB_ENV
+          echo "$RELEASE_BODY" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Trigger Test Deploy
+        run: |          
+          terminus env:deploy ${{ secrets.PANTHEON_SITE_NAME }}.test --note "${{ env.DEPLOY_TEST_MESSAGE }}"
+
+  deploy-to-live:
+    environment: Live # This environment requries manual approval
+    needs: deploy-to-test
+    permissions:
+      contents: read
+      deployments: write
+      pull-requests: read
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.release.tag_name }}  # Checkout the release tag (e.g., v1.0.0)
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "7.4"
+
+      - name: Install Terminus
+        uses: pantheon-systems/terminus-github-actions@v1
+        with:
+          pantheon-machine-token: ${{ secrets.PANTHEON_MACHINE_TOKEN }}
+          disable-cache: true
+
+      - name: Prepare deployment message
+        run: |
+          RELEASE_BODY=$(echo "${{ github.event.release.body }}")
+          echo "DEPLOY_TEST_MESSAGE<<EOF" >> $GITHUB_ENV
+          echo "$RELEASE_BODY" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Deploy to Live
+        run: |
+          terminus env:deploy ${{ secrets.PANTHEON_SITE_NAME }}.live --note "${{ env.DEPLOY_TEST_MESSAGE }}"
```

Comment:
Unless I'm missing something, the two jobs in this file are identical except for this last line. Can you restructure it somehow to reduce the duplication and share the identical config between the two jobs? Something like the `<<: *build_and_deploy` we're using the CircleCI config.yml to share common config?

Reply:
I tried to use "[composite action](https://docs.github.com/en/actions/sharing-automations/avoiding-duplication)" but it can't use secrets. Additionally, each job has to have its own "checkout" step which leaves only two common steps that are sandwiched between other steps that can't be shared.

Reply:
The checkout steps both look identical to me. I think you could probably use something like the `<<: *build_and_deploy` we're using in the CircleCI config.yml to define and insert a common block of config. See also  https://stackoverflow.com/questions/41063361/what-is-the-double-left-arrow-syntax-in-yaml-called-and-wheres-it-specifi

Reply:
Thanks! I didn't know this is a YAML feature called anchors. It looks like GitHub Actions doesn't support YAML anchor yet. It's currently [being promised for the second half of 2025](https://github.com/actions/runner/issues/1182).

Reply:
Oh okay. I hoped they supported that since it's a standard YAML feature. I guess we can update the action when they add support for that feature.